Source/compositorclient/RPI/ModeSet.cpp

```diff
@@ -410,11 +410,21 @@ struct gbm_surface* ModeSet::CreateRenderTarget(const uint32_t width, const uint
     return result;
 }
 
-void ModeSet::DestroyRenderTarget(struct gbm_surface* surface)
+void ModeSet::DestroyRenderTarget(struct BufferInfo& buffer)
 {
-    if (nullptr != surface)
+    if (nullptr != buffer._bo) 
     {
-        gbm_surface_destroy(surface);
+        if (static_cast<uint32_t>(~0) == buffer._id) {
+            drmModeRmFB(_fd, buffer._id);
+        }
+
+        gbm_surface_release_buffer(buffer._surface, buffer._bo);
+        buffer._bo = nullptr;
+    }
+
+    if (nullptr != buffer._surface)
+    {
+        gbm_surface_destroy(buffer._surface);
```

Comment:
All other resources associated with the surface are also released / freed. Is this intended behaviour?